drivers/UARTSerial.cpp

```diff
@@ -143,6 +160,10 @@ ssize_t UARTSerial::write(const void *buffer, size_t length)
         return 0;
     }
 
+    if (core_util_in_critical_section()) {
+        return write_unbuffered(buf_ptr, length);
```

Comment:
Should we advertise after this change, that printf from ISR is valid?

Reply:
Even after this, `printf` from ISR isn't valid - you'll hit a FILE mutex inside the C library.  What _may_ work is `write(STDERR_FILENO)`, which avoids the C library, but that in turn will depend on what exactly stderr is hooked to.

And we don't absolutely stop the user redirecting stderr to a `File`, eg using freopen.

The potential `write` from ISR is a "last resort" - it has nothing else it can reasonably do. I don't think we should absolutely require that write to stderr work from interrupts, but we can encourage people to try to use stderr devices that can handle this.